File: dev/20190704/k2works/api/rental/domain/rental.rb
module Domain
  class Rental
    attr_reader :movie, :days_rented
    def initialize(movie, days_rented)
      @movie, @days_rented = movie, days_rented
    end

    def charge
      @movie.charge(days_rented)
    end

    def frequent_renter_points
      (movie.price_code == Movie::NEW_RELEASE && days_rented > 1) ? 2 : 1
    end

  end
end

File: dev/20190710/k2works/api/video_rental/domain/rental.py


class Rental:
    def __init__(self, movie, days_rented):
        self.__movie = movie

        self.__days_rented = days_rented

    @property
    def movie(self):
        return self.__movie

    @property
    def days_rented(self):
        return self.__days_rented

    def charge(self):
        return self.movie.charge(self.days_rented)

    def frequent_renter_points(self):
        return self.movie.frequent_renter_points(self.days_rented)

